src/sardana/taurus/core/tango/sardana/macro.py

```diff
@@ -1166,13 +1166,84 @@ def ParamFactory(paramInfo):
         param = SingleParamNode(param=paramInfo)
     return param
 
+def check_member_node_simple(rest_raw, param_node, mem, rep, params_info_len):
+    for member_raw in rest_raw:
+        repeat_node = param_node.child(rep)
+        if repeat_node is None:
+            repeat_node = param_node.addRepeat()
+        member_node = repeat_node.child(mem)
+        if isinstance(member_node, RepeatParamNode):
+            params_info_nested = member_node.paramsInfo()
+            params_info_len_nested = len(params_info_nested)
+            rep_nested = 0; mem_nested = 0
+            check_member_node_simple(member_raw, member_node, mem_nested, rep_nested, params_info_len_nested)
+        else:
+            if not isinstance(member_raw, list):
+                member_node.setValue(str(member_raw))
+            else:
+                if len(member_raw) == 1:
+                    member_raw = member_raw[0]
+                    member_node.setValue(str(member_raw))
+        mem += 1
+        mem %= params_info_len
+        if mem == 0:
+            rep += 1
+  
+
+def check_interface(param_raw, param_node):
+    repeat_node = param_node.child(0)
+    for j, param in enumerate(param_raw):
+        if(isinstance(param,str)):
+            return 0
+        for k, par in enumerate(param):
+            member_node = repeat_node.child(k)
+            if isinstance(member_node, RepeatParamNode):
+                ret = check_interface(par, member_node)
+                if ret ==0:
+                    return ret
+    return 1
+            
+def check_member_node(param_raw, param_node):
+    for j, param in enumerate(param_raw):
+        repeat_node = param_node.child(j)
+        if repeat_node is None:
+            repeat_node = param_node.addRepeat()
+        for k, par in enumerate(param):
+            member_node = repeat_node.child(k)
+            if isinstance(member_node, RepeatParamNode):
+                check_member_node(par, member_node)
+            else:
+                try: # last parameters can have default values
+                    member_node.setValue(str(par))
+                except:
+                    pass
+       
+class Reflector(object):
+    def __getitem__(self, name):
+        return name
+  
+def recur_map(fun, data, keep_none=False):
+    """Recursive map. Similar to map, but maintains the list objects structure
+
+    :param fun: <callable> the same purpose as in map function
+    :param data: <object> the same purpose as in map function
+    :param keep_none: <bool> keep None elements without applying fun
+    """
+    if hasattr(data, "__iter__"):
+        return [recur_map(fun, elem, keep_none) for elem in data]
+    else:
+        if keep_none is True and data is None:
+            return data
+        else:
+            return fun(data)
+
+                
 def createMacroNode(macro_name, params_def, macro_params):
```

Comment:
I checked it again, that with list doesn't work, because we don't have a list. One can make it like it was, searching [ and ]. The problem is that passing this to the magic command doesn't work like
 it is, it is not simply moving it, one would have to study again every simple case.